components/forecast/Daily.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import styled from 'styled-components/native';
+import moment from 'moment';
+
+import weatherIcons from '../../constants/weathericons';
+import {ForecastTimestep} from '../../utils/locationforecast';
+
+function maxTemp(forecast: ForecastTimestep | undefined): number | undefined {
```

Comment:
As I understand it, the purpose of this code is to give max temperature for a single day, right?

If I read the code correctly, since `next_12_hours` is often `undefined`, the function will almost always return `undefined`. A more correct way could then be to do something like this (pseudo-code):

```
for t in [next_12_hour, next_6_hours, next_1_hours]:
  if t !== undefined: return t
```

This works, since `next_12_hours` is always at least as high as `next_6_hours` which, in turn, is at least as high as `next_1_hours`. 

..but that would in some cases return max temperature for only the coming 1-hour period for a day. I believe a better option would be to refactor, so you iterate over all time steps for a day (ie. 00:00:00, 01:00:00, 02:00:00, etc...), and then select the maximum max temperature from these.

Reply:
@vegardb. thanks.